src/main/java/com/shaft/cucumber/GuiActionSteps.java

```diff
@@ -0,0 +1,126 @@
+package com.shaft.cucumber;
+
+import com.shaft.gui.browser.BrowserActions;
+import com.shaft.gui.browser.BrowserFactory;
+import com.shaft.gui.element.ElementActions;
+import com.shaft.validation.Assertions;
+import io.cucumber.java.ParameterType;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+
+import java.util.Arrays;
+
+public class GuiActionSteps {
+    //TODO: handle shaft_engine listeners
+    private final ThreadLocal<WebDriver> driver = new ThreadLocal<>();
+
+    @Given("I Open the target browser")
+    public void BrowserFactory_getBrowser() {
+        driver.set(BrowserFactory.getBrowser());
+    }
+
+    @When("I Navigate to {string}")
+    public void BrowserActions_navigateToURL(String url) {
+        BrowserActions.navigateToURL(driver.get(), url);
+    }
+
+    @When("I Navigate to {string} and get redirected to {string}")
+    public void BrowserActions_navigateToURL(String url, String urlAfetrRedirection) {
+        BrowserActions.navigateToURL(driver.get(), url, urlAfetrRedirection);
+    }
+
+    @When("I Close the current window")
+    public void BrowserActions_closeCurrentWindow() {
+        BrowserActions.closeCurrentWindow(driver.get());
+    }
+
+    @ParameterType(".*")
+    public LocatorType locatorType(String locatorName) {
+        return getLocatorTypeFromName(locatorName);
+    }
+
+    @When("I Type {string} into the element found by {locatorType}: {string}")
+    public void ElementActions_type(String text, LocatorType locatorType, String locatorValue) {
+        ElementActions.type(driver.get(), getLocatorFromTypeAndValue(locatorType, locatorValue), text);
+    }
+
+    /**
+     * Sends a keypress to the target element. Supported keys are: ENTER, RETURN, TAB
+     *
+     * @param key
+     * @param locatorType
+     * @param locatorValue
+     */
+    @When("I Press the {word} key into the element found by {locatorType}: {string}")
+    public void ElementActions_keyPress(String key, LocatorType locatorType, String locatorValue) {
+        ElementActions.keyPress(driver.get(), getLocatorFromTypeAndValue(locatorType, locatorValue), key);
+    }
+
+    @When("I Click the element found by {locatorType}: {string}")
+    public void ElementActions_click(LocatorType locatorType, String locatorValue) {
+        ElementActions.click(driver.get(), getLocatorFromTypeAndValue(locatorType, locatorValue));
+    }
+
+    @Then("I Assert that the {word} attribute of the element found by {locatorType}: {string}, should be {string}")
+    public void Assertions_assertElementAttribute(String elementAttribute, LocatorType locatorType, String locatorValue, String expectedValue) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The instance method name 'Assertions_assertElementAttribute' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/mohab.mohieeldeen/SHAFT_ENGINE/pullRequest?prid=5707958)